### PR TITLE
New CLI parameter interface for block sealing

### LINKED ISSUE #1517: Change CLI parameters for sealing in preparation for new interval sealing mode
--sealing <SEALING>
          The sealing mode
          
          [default: instant]

          Possible values:
          - instant:
            Author a block immediately upon receiving a transaction into the transaction pool
          - manual:
            Author a block upon receiving an RPC command
          - interval:
            Author blocks at a regular interval specified in seconds

      --sealing-interval <SECONDS>
          The interval in seconds
          
          [default: 120]

      --sealing-allow-empty-blocks
          Allows empty blocks to be created when sealing in manual and interval modes

### PR DESCRIPTION
**Pull request by rlaferla**
# Goal
The goal of this PR is to provide more robust CLI parameters for sealing modes.

Closes #1517

### CODE REVIEW

File: node/cli/src/cli.rs [hunk 1]
```diff
@@ -79,19 +84,23 @@ pub struct Cli {
 	/// Instant block sealing
 	/// Blocks are triggered to be formed each time a transaction hits the validated transaction pool
 	/// Empty blocks can also be formed using the `engine_createBlock` RPC
-	///
-	/// Can only be used with frequency-no-relay and frequency-rococo-local feature flags
-	#[cfg(any(feature = "frequency-no-relay", feature = "frequency-rococo-local"))]
-	#[clap(long = "instant-sealing")]
-	pub instant_sealing: bool,
```
Comment (wilwade) [inline node/cli/src/cli.rs:86]: Is there a way with clap to leave a message that the option has changed?

Aka if someone after this runs ./frequency --instant-sealing it wouldn't run, but would report that it has changed to be --sealing=instant ?

File: node/cli/src/run_as_localchain.rs [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+	// Temporary error message
+	if cli.sealing == SealingMode::Interval {
+		return Result::<(), _>::Err(sc_cli::Error::Input(String::from(
+			"Interval sealing is not yet implemented.",
+		)))
+	}
+
+	let current_time = Local::now();
+	let formatted_time = current_time.format("%Y-%m-%d %H:%M:%S").to_string();
+	println!("{} NO-RELAY MODE with {} sealing", formatted_time, cli.sealing);
```
Comment (wilwade) [inline node/cli/src/run_as_localchain.rs:16]: Don't we get the time part for free when we use the `log::info!` logging?

**Review by wilwade — COMMENTED** [review]
One question and I think a log to switch to the info macro.

File: node/cli/src/cli.rs (hunk 1 above)
Comment (rlaferla) [reply node/cli/src/cli.rs:86]: I will check to see if it has support for deprecated parameters.  If not, I could trap them and display a warning message.

File: node/cli/src/run_as_localchain.rs (hunk 2 above)
Comment (rlaferla) [reply node/cli/src/run_as_localchain.rs:16]: Normally but info level logs are being filtered so it doesn't display.  Hence, the workaround.  If I could only modify the runner to append additional info.

File: node/cli/src/cli.rs (hunk 1 above)
Comment (rlaferla) [reply node/cli/src/cli.rs:86]: Fixed.

File: node/cli/Cargo.toml [hunk 3]
```diff
@@ -11,6 +11,7 @@ version = "0.0.0"
 [dependencies]
 clap = { version = "4.0.17", features = ["derive"] }
 codec = { package = "parity-scale-codec", version = "3.1.5" }
+chrono = { version = "0.4.24" }
```
Comment (wilwade) [inline node/cli/Cargo.toml:14]: I think you can drop this dep now.

**Review by wilwade — APPROVED** [review]
🎉

**Review by mattheworris — CHANGES_REQUESTED** [review]
Looks good! One update to `--instant-sealing` missed in `init.sh` for `start-frequency-native` when my PR was merged.

**Review by mattheworris — APPROVED** [review]

**Pull request merged**